pkgs/stdenv/generic/setup.sh

```diff
@@ -334,36 +334,44 @@ appendToVar() {
     fi
 }
 
-# Accumulate into `flagsArray` the flags from the named variables.
+# Accumulate flags from the named variables $2+ into the array $1.
 #
-# If __structuredAttrs, the variables are all treated as arrays
-# and simply concatenated onto `flagsArray`.
-#
-# If not __structuredAttrs, then:
-#   * Each variable is treated as a string, and split on whitespace;
-#   * except variables whose names end in "Array", which are treated
-#     as arrays.
-_accumFlagsArray() {
-    local name
-    if [ -n "$__structuredAttrs" ]; then
-        for name in "$@"; do
-            local -n nameref="$name"
-            flagsArray+=( ${nameref+"${nameref[@]}"} )
-        done
-    else
-        for name in "$@"; do
+# Arrays are simply concatenated, strings are split on whitespace.
+concatTo() {
+    local -n targetref="$1"; shift
+    local name type
+    for name in "$@"; do
+        if type=$(declare -p "$name" 2> /dev/null); then
             local -n nameref="$name"
-            case "$name" in
-                *Array)
-                    # shellcheck disable=SC2206
-                    flagsArray+=( ${nameref+"${nameref[@]}"} ) ;;
+            case "${type#* }" in
+                -A*|-a*)
+                    targetref+=( "${nameref[@]}" ) ;;
```

Comment:
Maybe need to document and motivate what happens to associative arrays. If `nameref` is an associative array, this only adds the values? It's seemingly expected that `targetref` is not associative?